pkg/logs/input/docker/launcher.go

```diff
@@ -40,7 +41,8 @@ type Launcher struct {
 	erroredContainerID chan string
 	lock               *sync.Mutex
 	collectAllSource   *config.LogSource
-	readTimeout        time.Duration // client read timeout to set on the created tailer
+	readTimeout        time.Duration               // client read timeout to set on the created tailer
+	serviceFunc        func(string, string) string // serviceFunc gets the service name from the tagger, it is in a separate field for testing purpose
```

Comment:
nit: service is a very overloaded word in the logs agent, could we rename `serviceFunc` to `serviceNameFunc` everywhere maybe? Just to semantically separate it a bit from `service.Service`.